ext/storage_sources/azure_store/test/test_azure_connection.cpp

```diff
@@ -31,4 +31,26 @@
 
 #include "azure_connection.h"
 
-TEST_CASE("testing class azure_connection", "azure_connection") {}
+TEST_CASE("testing class azure_connection", "azure_connection")
+{
+    SECTION("List blobs under the test container.") {}
+
+    SECTION("Test object_exists.")
+    {
+        azure_connection conn = azure_connection("myblobcontainer1", "object_exist_test_prefix");
+        bool object_exists = false;
+
+        // Check for a non-existant object in the container.
+        REQUIRE(conn.object_exists("test.txt", object_exists) == 0);
+        REQUIRE(!object_exists);
+
+        // Check for the newly added object in the container.
+        conn.put_object(
+          "test.txt", "/home/ubuntu/wiredtiger/ext/storage_sources/azure_store/test.txt");
+        REQUIRE(conn.object_exists("test.txt", object_exists) == 0);
+        REQUIRE(object_exists);
+
+        // Delete test case object.
+        REQUIRE(conn.delete_object("test.txt") == 0);
```

Comment:
If we don't get to here (if say, something throws an exception) will this leave the thing around, costing money, forever?

Reply:
The way we've structured the unit tests is that we're using `#define CATCH_CONFIG_MAIN`, so all the tests wrapped in `REQUIRE` will run irrespective of earlier test outcomes. The only situation where the object would stay in the cloud is if the `delete_object()` functionality wasn't implemented properly. As of right now, I can see that `delete_object` is working fine through the Azure Portal.

Reply:
I don't think that's quite true - the catch2 docs say "The REQUIRE family of macros tests an expression and aborts the test case if it fails", so an earlier `REQUIRE` tripping will abort the test and not run this. As a demo, you could add `REQUIRE(0 == 1)` above this, and I suspect `test.txt` would not be deleted.

Reply:
opps I misunderstood the doc. I'll replace `REQUIRE` with `CHECK` for all test cases except the `delete_object` test so that test execution continues even if an assertion fails. Thanks for clarifying this :)

Reply:
I think there's a better way - we can use RAII semantics here by having a small `class AzureObjectWrapper` or similar. It's constructor could take a name, and the destructor would delete the object. That we we wouldn't have to write the tests in a certain way to make sure objects don't get left lying around.

Reply:
that's a great suggestion, we'll discuss implementing this over a meeting tomorrow so we get everyone on the same page.